src/cli/api/local_api.py

```diff
@@ -124,18 +131,58 @@ def list_applications(self) -> List[ApplicationType]:
 
     # Todo: Update confluence scenario diagram since application_unique() and structure_valid() are swapped
     def is_application_valid(self, application: str) -> Tuple[bool, str]:
+        """
+        Function that checks if:
+        - The application is valid by validating if it is unique
+        - The file and directory structure is correct
+        - The json files in the config directory contain valid json
+        - The json files passes against schema validation
+
+        returns:
+            Returns True if all validations passes
+            Returns False if one of the validations throws an Exception
+        """
+
         if self.__is_application_unique(application) and \
            self.__is_structure_valid(application) and \
            self.__is_config_valid(application):
             return True, "Valid"
 
         return False, "Invalid"
 
+    def __is_config_valid(self, application: str) -> bool:
+        # Validate if json string is correct and validate against json schema's
+        schema_app_path = Path(BASE_DIR + "/schema/applications")
+
+        validate_json(self.app_config / "application.json", schema_app_path / "appconfig_app.json")
+        validate_json(self.app_config / "network.json", schema_app_path / "appconfig_network.json")
+        validate_json(self.app_config / "network.json", schema_app_path / "appconfig_result.json")
+
+        return True
+
     def __is_structure_valid(self, application: str) -> bool:
-        pass
+        # Validate that the config, src and MANIFEST.ini structure is correct
+        if not os.path.exists(self.app_config) or \
+           not os.path.exists(self.app_source) or \
+           not os.path.isfile(Path.cwd() / "MANIFEST.ini"):
+            raise ApplicationDirectoryNotComplete(Path.cwd())
+
+        # Check if the config directory is complete
+        if not os.path.isfile(self.app_config / "application.json") or \
+           not os.path.isfile(self.app_config / "network.json") or \
+           not os.path.isfile(self.app_config / "result.json"):
+            raise ApplicationConfigNotComplete(self.app_config)
+
+        # Check if there is are at least two python files in the src directory
```

Comment:
I think this check should check if app_<role1>.py ..  app_<rolen>.py exist. Just as we did when the application was created. Because appsource the filenames are specified by the rolename. The rolenames you can get from multiple sources (application.json and network.json)